src/mjolnir/graphfilter.cc

```diff
@@ -27,10 +29,157 @@ uint32_t n_original_nodes = 0;
 uint32_t n_filtered_edges = 0;
 uint32_t n_filtered_nodes = 0;
 uint32_t can_aggregate = 0;
+uint32_t aggregated = 0;
 
 // Group wheelchair and pedestrian access together
 constexpr uint32_t kAllPedestrianAccess = (kPedestrianAccess | kWheelchairAccess);
 
+struct EdgeId {
+  uint64_t way_id;
+  GraphId graph_id;
+};
+
+bool CanAggregate(const DirectedEdge* de) {
+  if (de->start_restriction() || de->part_of_complex_restriction() || de->end_restriction() ||
+      de->restrictions() || de->traffic_signal() || de->access_restriction()) {
+    return false;
+  }
+  return true;
+}
+
+bool ExpandFromNode(GraphReader& reader,
```

Comment:
these pain me a bit. we have so many small-ish expansion algos distributed across the code base, most probably doing mostly the same thing. nothing to take action right now, but it'd  be really nice if we could consolidate them a bit more "in the future"©

Reply:
ha agree